design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,231 @@
+
+However, Crossplane is not an arbitrary Kubernetes cluster to run typical compute and storage workloads.
+Instead, it is a centralized control plane intended to manage resources external to the control plane (e.g. cloud provider services) and intelligently schedule compute workloads into remote clusters to consume these services.
+There is not really a need for Stacks being installed into the control plane to request arbitrary permissions.
+Crossplane can be more opinionated about the types of permissions granted within its control plane and lock things down in a more secure way.
+
+Futhermore, stacks are currently scoped to a single type that is used in all scenarios and by all personas.
+All new types of functionality, from an end user wanting to install a simple application to an administrator wanting to install support for low level infrastructure types, are all treated the same.
+They have the same approach to security and access control and the artifacts that they install have the same scope of impact to the control plane.
+This one size fits all approach is not ideal though, the multiple scenarios and personas need finer grain control around security, access, and isolation.
+
+This design doc proposes a new approach for the Crossplane Stack Manager (SM) that will support a smaller set of security permissions and increased control for the installation of stacks.
+
+## Design Summary
+
+The key differences in this design update are in the **scope** of the permissions that stacks can be granted within the control plane.
+
+* **Cluster Stacks** will be given permissions to watch, access, and create resources at a **cluster level**, i.e. across **all namespaces**.
+* **Stacks** will be given permissions only within a **single namespace**.
+* All stacks will be granted permissions to a small subset of core Kubernetes resources as well as access to any CRDs they own and depend on.
+
+This means that stacks should not be able to directly affect any resources outside of the specific namespace they are installed into, while cluster stacks operate at a higher level of privilege and can affect the entire cluster.
+This is a familiar concept to the Kubernetes permission scoping model of `ClusterRole` and `Role`.
+
+### CRD Scoping
+
+To avoid a common point of confusion, let's review the available scoping of CRDs.
+Custom resource **instances** can be scoped to either a single namespace or to the entire cluster.
+It's important to note that this refers to the instances, not the CRD type definitions.
+From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition):
+
+> CustomResourceDefinitions themselves are non-namespaced and are available to all namespaces
+
+Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
```

Comment:
I understand the concept now, but I'm not sure why I needed to understand it yet. Would it make sense to include the reason that it's important?

Reply:
It's a common point of confusion in the Kubernetes ecosystem and this section is just conveying a fact to level set.  The reason is to "avoid a common point of confusion", as the section starts.  I honestly don't think any more explanation is needed to the reader.